pkg/v1/internal/estargz/estargz.go

```diff
@@ -31,7 +34,7 @@ import (
 //
 // Refer to estargz for the options:
 // https://pkg.go.dev/github.com/containerd/stargz-snapshotter@v0.2.0/estargz#Option
-func ReadCloser(r io.ReadCloser, opts ...estargz.Option) (io.ReadCloser, v1.Hash, error) {
+func ReadCloser(r io.ReadCloser, opts ...estargz.Option) (*estargz.Blob, v1.Hash, error) {
```

Comment:
Hmm... so all we're doing here is actually implementing `func(io.ReadCloser) (io.SectionReader, error)`...

Do we really need this? If we want to keep it, can we update the return signature to match `estargz.Build`? I'd honestly rather they change their interface and take advantage of any implemented interfaces, and then we just drop this entirely.

Reply:
This is a projection to our idioms, e.g. taking `io.ReadCloser` and returning `v1.Hash` (they cannot adopt the latter without a dependency cycle, but we could switch to using `go-digest` everywhere).

As #876 evolves, perhaps the necessity of this will diminish, but I think it's a useful place for us to mitigate things in the interim, and since it's internal there's effectively zero risk that we can't remove it.  🤷